### ISSUE TITLE
max_unpooling_layer is abstaract

### ISSUE THREAD

**Issue by pi-null-mezon** [tiny-dnn/tiny-dnn#279]
Hi once again. Making experiments with tiny-dnn-0.1.1 I have faced with another compilation error when try to add max_unpooling_layer to the sequential network. MSVC2013x86 and MSVC2015x86 both says:  'tiny_cnn::max_unpooling_layer<tiny_cnn::activation::relu>' : cannot instantiate abstract class due to following members: 'void tiny_cnn::layer::forward_propagation...

**Comment by Wangyida** [comment]
@pi-null-mezon Could you have a try on head of a [former version](https://github.com/Wangyida/tiny-cnn/tree/quantization) and see if there are also such issue?

**Comment by pi-null-mezon** [comment]
Former version does not throw compile time error. But max_unpooling activation looks strange - it simply blinks from black to white with small bar at left top corner:
![image](https://cloud.githubusercontent.com/assets/5991537/17690437/57e8636a-6397-11e6-840d-36ee52bd0ea5.png)
![image](https://cloud.githubusercontent.com/assets/5991537/17690457/6cd68108-6397-11e6-9019-4b024d0b0767.png)

**Comment by Wangyida** [comment]
@pi-null-mezon Max unpool is set on abstract layer now and you can have a try on the ave-unpool layer instead as in the example. Examples are defined [here](https://github.com/Wangyida/tiny-cnn/blob/quantization/examples/deconv/train.cpp#L116-L121), you could also have a reference on the [sorce codes](https://github.com/Wangyida/tiny-cnn/blob/quantization/tiny_cnn/layers/average_unpooling_layer.h)

**Comment by pi-null-mezon** [comment]
Average unpooling layer works well. It little bit confusing that abstract layer have name "max_unpooling_layer", maybe better name for him will be "unpooling_layer"?

**Comment by Wangyida** [comment]
@pi-null-mezon Yes, that's the initial idea for this layer that max unpooling layer should have corresponding patch-wise max-Index as another input also.
